### PR TITLE
hide grid after selecting a tab from history view

### PR DESCRIPTION
**Pull request by chassu**
### How was this tested?
- [x] I tested this manually

### Screenshots and screen recordings
before:

https://user-images.githubusercontent.com/87332917/164108913-575826df-33e4-4e38-abdb-54605aa4e1ba.mp4

after:

https://user-images.githubusercontent.com/87332917/164108937-c8f5773a-ad86-4298-ac98-2270def9572d.mp4

### CODE REVIEW

**Comment by darinwf** [comment]
Hmm... tapping on an item in History should use createOrSwitchToTab and **bring the selected tab to the foreground**. It is perhaps a regression from the new SwiftUI-based History UI that that isn't happening now.

**Comment by darinwf** [comment]
Something like this:
```
diff --git a/Client/Frontend/Browser/BrowserViewController.swift b/Client/Frontend/Browser/BrowserViewController.swift
index d9fd2f0e9..6a4da31ed 100644
--- a/Client/Frontend/Browser/BrowserViewController.swift
+++ b/Client/Frontend/Browser/BrowserViewController.swift
@@ -1494,6 +1494,10 @@ extension BrowserViewController {
         }
 
         updateInZeroQuery(selected?.url as URL?)
+
+        if selected != nil {
+            hideCardGrid(withAnimation: false)
+        }
     }
 
     func getSceneDelegate() -> SceneDelegate? {
```

**Review by darinwf — CHANGES_REQUESTED** [review]
See comment above on suggested change.

**Comment by chassu** [comment]
https://user-images.githubusercontent.com/87332917/165200003-8227051f-2c76-413c-982f-8e487d66a3b1.mp4

**Comment by chassu** [comment]
looks the that fix alone is causing the normal card transition animation to fail. will investigate it.

File: Client/Frontend/Site/SiteRowView.swift [hunk 1]
```diff
@@ -32,6 +33,7 @@ struct SiteRowView: View {
             Button {
                 action()
                 openURL(site.url)
+                browserModel.hideGridWithNoAnimation()
```
Comment (darinwf) [inline Client/Frontend/Site/SiteRowView.swift:36]: Can this be moved within the `openURL` handler? I feel like it is a lot for `SiteRowView` to have to know about hiding the grid.

Comment (darinwf) [reply Client/Frontend/Site/SiteRowView.swift:36]: Yeah, `HistoryPanelViewController` seems like a better spot for this. WDYT?

Comment (chassu) [reply Client/Frontend/Site/SiteRowView.swift]: makes sense to me!

**Review by darinwf — APPROVED** [review]

**Comment by darinwf** [comment]
When you commit be sure to update the commit log entry accordingly since this PR evolved a lot since the start :)

**Pull request merged**